src/plugins/mergeCells/test/mergeCells.e2e.js

```diff
@@ -1311,4 +1311,60 @@ describe('MergeCells', () => {
       expect(plugin.mergedCellsCollection.mergedCells.length).toEqual(2);
     });
   });
+
+  describe('cooperation with the `Autofill` plugin', () => {
+    it('should add new merged areas when dragged the merged cell', () => {
+      handsontable({
+        data: Handsontable.helper.createSpreadsheetObjectData(5, 5),
+        mergeCells: [
+          { row: 0, col: 0, rowspan: 2, colspan: 2 }
+        ],
+        fillHandle: true
+      });
+
+      selectCell(0, 0);
+
+      // Dragging merged cell one level down.
+      spec().$container.find('.wtBorder.current.corner').simulate('mousedown');
+      spec().$container.find('tr:eq(2) td:eq(0)').simulate('mouseover');
+      spec().$container.find('tr:eq(2) td:eq(0)').simulate('mouseup');
+
+      // First merged cell.
+      expect(spec().$container.find('tr:eq(0) td:eq(0)')[0].offsetWidth).toBe(100);
+      expect(spec().$container.find('tr:eq(0) td:eq(0)')[0].offsetHeight).toBe(47);
+      expect(getCell(0, 1).innerText).toBe('A1');
+      expect(getDataAtCell(0, 0)).toBe('A1');
+      // Already populated merged cell.
+      expect(spec().$container.find('tr:eq(2) td:eq(0)')[0].offsetWidth).toBe(100);
+      expect(spec().$container.find('tr:eq(2) td:eq(0)')[0].offsetHeight).toBe(46);
+      expect(getCell(2, 1).innerText).toBe('A1');
+      expect(getDataAtCell(2, 0)).toBe('A1');
+
+      selectCell(0, 0);
+
+      // Dragging merged cell one level right.
+      spec().$container.find('.wtBorder.current.corner').simulate('mousedown');
+      spec().$container.find('tr:eq(0) td:eq(2)').simulate('mouseover');
+      spec().$container.find('tr:eq(0) td:eq(2)').simulate('mouseup');
+
+      // First merged cell.
+      expect(spec().$container.find('tr:eq(0) td:eq(0)')[0].offsetWidth).toBe(100);
+      expect(spec().$container.find('tr:eq(0) td:eq(0)')[0].offsetHeight).toBe(47);
+      expect(getCell(0, 1).innerText).toBe('A1');
+      expect(getDataAtCell(0, 0)).toBe('A1');
+      // Previously populated merged cell.
+      expect(spec().$container.find('tr:eq(2) td:eq(0)')[0].offsetWidth).toBe(100);
+      expect(spec().$container.find('tr:eq(2) td:eq(0)')[0].offsetHeight).toBe(46);
+      expect(getCell(2, 1).innerText).toBe('A1');
+      expect(getDataAtCell(2, 0)).toBe('A1');
+      // Already populated merged cell.
+      expect(spec().$container.find('tr:eq(0) td:eq(2)')[0].offsetWidth).toBe(100);
+      expect(spec().$container.find('tr:eq(0) td:eq(2)')[0].offsetHeight).toBe(47);
+      expect(getCell(0, 3).innerText).toBe('A1');
+      expect(getDataAtCell(0, 2)).toBe('A1');
+
+      expect($(getHtCore())[0].offsetWidth).toBe(5 * 50);
+      expect($(getHtCore())[0].offsetHeight).toBe(24 + (4 * 23)); // First row is 1px higher than others.
```

Comment:
What would you like to check from the second row? In some cases, the container hasn't proper size.

I'm checking the height of the **full table**, not just the sum of heights of all cells. In such a case, it would be a kind of workaround for me.

Reply:
Oh right, my bad, sorry. Nothing to do here then.